operator/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -747,6 +748,126 @@ func (s *RedpandaControllerSuite) TestScaling() {
 	s.deleteAndWait(rp)
 }
 
+func (s *RedpandaControllerSuite) TestNodePools() {
+	isStable := func(checkStable bool) func(obj client.Object, err error) (bool, error) {
+		return func(obj client.Object, err error) (bool, error) {
+			if err != nil {
+				return false, err
+			}
+			var conditions []metav1.Condition
+			var stable string
+			switch typ := obj.(type) {
+			case *redpandav1alpha2.Redpanda:
+				conditions = typ.Status.Conditions
+				stable = statuses.ClusterStable
+			case *redpandav1alpha2.NodePool:
+				conditions = typ.Status.Conditions
+				stable = statuses.NodePoolStable
+			default:
+				return false, errors.New("unsupported type")
+			}
+			status := metav1.ConditionTrue
+			if !checkStable {
+				status = metav1.ConditionFalse
+			}
+			for _, cond := range conditions {
+				if cond.Type == stable {
+					return cond.ObservedGeneration == obj.GetGeneration() && cond.Status == status, nil
+				}
+			}
+			return false, nil
+		}
+	}
+
+	rp := s.minimalRP()
+
+	// start with one broker and no nodepools
+	s.applyAndWaitFor(isStable(true), rp)
+
+	// add another broker via a nodepool.
+	pool := s.minimalNodePool(rp)
+	s.applyAndWaitFor(isStable(true), pool)
+
+	s.deleteAndWait(rp)
+
+	// we should go unstable due to an unhealthy broker
+	s.waitFor(pool, isStable(false))
+	// and eventually we should fully unbind
+	s.waitFor(pool, func(o client.Object, err error) (bool, error) {
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range o.(*redpandav1alpha2.NodePool).Status.Conditions {
+			if cond.Type == statuses.NodePoolBound {
+				return cond.ObservedGeneration == o.GetGeneration() && cond.Status == metav1.ConditionFalse, nil
+			}
+		}
+		return false, nil
+	})
+	s.deleteAndWait(pool)
+}
+
+func (s *RedpandaControllerSuite) TestNodePoolsBlueGreen() {
+	isStable := func(checkStable bool) func(obj client.Object, err error) (bool, error) {
+		return func(obj client.Object, err error) (bool, error) {
+			if err != nil {
+				return false, err
+			}
+			var conditions []metav1.Condition
+			var stable string
+			switch typ := obj.(type) {
+			case *redpandav1alpha2.Redpanda:
+				conditions = typ.Status.Conditions
+				stable = statuses.ClusterStable
+			case *redpandav1alpha2.NodePool:
+				conditions = typ.Status.Conditions
+				stable = statuses.NodePoolStable
+			default:
+				return false, errors.New("unsupported type")
+			}
+			status := metav1.ConditionTrue
+			if !checkStable {
+				status = metav1.ConditionFalse
+			}
+			for _, cond := range conditions {
+				if cond.Type == stable {
+					return cond.ObservedGeneration == obj.GetGeneration() && cond.Status == status, nil
+				}
+			}
+			return false, nil
+		}
+	}
+
+	rp := s.minimalRP()
+
+	// set the default pool's replicas to 0
+	rp.Spec.ClusterSpec.Statefulset.Replicas = ptr.To(0)
+
+	// start with one broker and no nodepools
+	s.applyAndWaitFor(isStable(true), rp)
```

Comment:
~Is there a typo in here or do we force deployments to have at least 1 replica?~

I saw the case above. This comment should be updated but I'm not sure I agree with or understand this behavior.